### PR TITLE
misc refactoring and student improvements

### LINKED ISSUE #238: As a student, I want to be able to view my current GPA

### LINKED ISSUE #326: deprecate NameColumn in favor of order_by in attributes
name columns are not reliable. They are (were) implemented ith NameColumn which tries to sort by "name_sort" column, but that only works in one or two places. Bleah.

### PR DESCRIPTION
**Pull request by bjmckenz**
fixes #238 by moving stuff to Student History -- and adds summary columns below history table
fixes #326, #328 by refactoring some stuff away.

### CODE REVIEW

File: student-info-system/sis/test_data/create_40_sectionstudent.py [hunk 1]
```diff
@@ -40,7 +40,7 @@ def createData():
             # sort of interesting...
```
Comment (jeffreicher) [inline student-info-system/sis/test_data/create_40_sectionstudent.py:40]: @bjmckenz is this being saved for later?

File: student-info-system/sis/tests/tests_model.py [hunk 2]
```diff
@@ -726,13 +725,14 @@ def test_admin_role(self):
...
-        usr = User.objects.create(username='foo')
-        prof = usr.profile
-        self.assertEqual(prof.role, Profile.ACCESS_NONE)
-        users = Profile.objects.all()
-        self.assertEqual(users.count(), 1)
-        users_ann = User.annotated()
-        self.assertEqual(users_ann.count(), 0)
-        prof.delete()
-        usr.delete()
+    # test is now invalid given retirement of annotated()
```
Comment (jeffreicher) [inline student-info-system/sis/tests/tests_model.py]: @bjmckenz this as well?

**Review by jeffreicher — COMMENTED** [review]
code review looks good - minor commented out code question

**Review by jeffreicher — APPROVED** [review]
pulled branch - looks good!

File: student-info-system/sis/test_data/create_40_sectionstudent.py (hunk 1 above)
Comment (bjmckenz) [reply student-info-system/sis/test_data/create_40_sectionstudent.py:40]: yes. It's a measure of how many courses are denied to students by not meeting prereqs. In the real world it should decline. I was watching it, and thinking if I could select courses better to be more representative of real data.

File: student-info-system/sis/tests/tests_model.py (hunk 2 above)
Comment (bjmckenz) [reply student-info-system/sis/tests/tests_model.py]: yeah, wanted to commit a reason for it being invalid. Not that important. and it's gone....

**Pull request merged**